extensions/typescript-language-features/src/features/referencesCodeLensOnAllFunctions.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as Proto from '../protocol';
+import * as PConst from '../protocol.const';
+import { CachedResponse } from '../tsServer/cachedResponse';
+import { ITypeScriptServiceClient } from '../typescriptService';
+import { ConfigurationDependentRegistration } from '../utils/dependentRegistration';
+import { TypeScriptReferencesCodeLensProvider } from './referencesCodeLens';
+import { getSymbolRange } from './baseCodeLensProvider';
+
+class TypeScriptReferencesCodeLensOnAllFunctionsProvider extends TypeScriptReferencesCodeLensProvider {
```

Comment:
I don't think this should be it's own class since the change adds extra functionality for the existing `ReferencesCodeLensProvider`.

Can you try checking `referencesCodeLens.showOnAllFunctions ` inside the class instead?

Reply:
@mjbvz 

Thank you for your comment.

When I implement the configuration dependent registration (like `referencesCodeLens.showOnAllFunctions`), 
I thought I needed to create `register` function as below, then each configuration should have its own class to use `vscode.languages.registerCodeLensProvider`;

https://github.com/microsoft/vscode/blob/1d7d80b23f65f637adadba59e4c9055e3d54a727/extensions/typescript-language-features/src/features/referencesCodeLens.ts#L93-L103

Please let me know if you have any smarter ideas to add a new setting in `TypeScriptReferencesCodeLensProvider` class. 🙇

Reply:
In  `TypeScriptReferencesCodeLensProvider`, you can read configuration values like this:

https://github.com/microsoft/vscode/blob/a188b905c6c079cc5b1255b0f61345e0d8c0f74c/extensions/typescript-language-features/src/features/completions.ts#L318-L319

Try reading the value for `referencesCodeLens.showOnAllFunctions` using this to change the behavior. To do this, you will need to pass the `modeId` (which is either `typescript` or `javascript`) to the `TypeScriptReferencesCodeLensProvider` class